staging/src/k8s.io/apimachinery/pkg/util/httpstream/spdy/upgrade.go

```diff
@@ -63,8 +65,12 @@ func (w *connWrapper) Close() error {
 // NewResponseUpgrader returns a new httpstream.ResponseUpgrader that is
 // capable of upgrading HTTP responses using SPDY/3.1 via the
 // spdystream package.
-func NewResponseUpgrader() httpstream.ResponseUpgrader {
-	return responseUpgrader{}
+//
+// If connPingPeriod is non-zero, for each incoming connection a background
+// goroutine will send periodic Ping frames to the server. Use this to keep
+// idle connections through certain load balancers alive longer.
+func NewResponseUpgrader(connPingPeriod time.Duration) httpstream.ResponseUpgrader {
```

Comment:
Here too, don't change the public signature.

Reply:
done